src/components/slide_in_panel.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { Motion, spring } from 'react-motion';
+import * as classNames from 'classnames';
+
+import CloseButton from "./close_button";
+
+export interface SlideInPanelProps {
+  /** Text to be displayed in the panel header */
+  headerTitle: string;
+  /** The panel will be open when true */
+  isOpen: boolean;
+  /** Called when the background overlay is clicked */
+  onOverlayClickHandler?: Function;
+  /** Width of panel. Include unit */
+  panelWidth?: string;
+}
+
+export class SlideInPanel extends React.Component<SlideInPanelProps, {}> {
+
+  _handleOverlayClick(e: React.MouseEvent<HTMLDivElement>): void {
+    const { onOverlayClickHandler } = this.props;
+
+    e.stopPropagation();
+
+    if (onOverlayClickHandler) {
+      onOverlayClickHandler();
+    }
+  }
+
+  render(): JSX.Element {
+    const { headerTitle, isOpen, panelWidth, children } = this.props;
+
+    const motion = { stiffness: 450, damping: 50 };
+    const classes = classNames('slide-in-panel', { isOpen });
+
+    return <div className={classes} onClick={(e) => this._handleOverlayClick(e)}>
+
+      <Motion style={{ x: spring(isOpen ? 0 : 100, motion) }}>
```

Comment:
Not sure how the motion is calculated, but it seems a bit jittery. Is it pure js or css? Css would be better as that's hardware accelerated.

Reply:
This does use js. Is the Slide out container component also jittery for you?

Reply:
Yup.